pageserver/src/layered_repository.rs

```diff
@@ -1128,8 +1128,6 @@ impl Timeline for LayeredTimeline {
 
     /// Look up the value with the given a key
     fn get(&self, key: Key, lsn: Lsn) -> Result<Bytes> {
-        debug_assert!(lsn <= self.get_last_record_lsn());
```

Comment:
hmm why do we need to remove this?

Reply:
Because I take `lsn` from  `DatadirModification` and it can be larger than `last_record_lsn` at the moment of adding modification because `last_record_lsn` is advanced by `DatadirModification::commit`.

Reply:
R u referring to this line: https://github.com/neondatabase/neon/pull/2094/files#diff-bf5842dbfabca9cedacf26fe7e9be45370448b19b5d0191e540b1d11739f9ca1R946.

I found that previously it was
```
let last_lsn = self.timeline.get_last_record_lsn();
```

Any reasons for this change? Look like `lsn` of `DatadirModification` is also made public for this

Reply:
For the same reason: lat_record_lsn is advanced after all modifications are done. So if we use `last_record_lsn` in relation size entry, then it will be smaller than expected LSN value.

Reply:
Got it. In that way, https://github.com/neondatabase/neon/pull/2094/files#diff-bf5842dbfabca9cedacf26fe7e9be45370448b19b5d0191e540b1d11739f9ca1R927 should also be changed right?

Reply:
This is read only method: no reference to `DatadirModification`.
So `getlast_record_lsn` is the only what to obtain current LSN.